### PR TITLE
Improvement/refactor discovery related stuff

### PR DESCRIPTION
**Pull request by tadovas**
One biggest pains, which asked for refactoring:
1. Splitted global MysteriumClient interface into small ones and moved to callers - no more dependencies to mysterium client which is renamed to mysterium api (access to our central discovery service)
2. Introduced market package with subpackages - top root package defines proposal related stuff (all shared structures like Proposal, Location, Service and so on), concrete functionalities have own subpackages (metrics, central api access, proposal lifecycle component)
3. Basically a lot of files touched because of proposal moving to different package

Bonus: almost all packages are golint'ed :)

Question: which sounds better? proposals.ServiceProposal or market.ServiceProposal ? In second case I will move proposals up one package (to market)

### CODE REVIEW

File: market/mysterium/mysterium_api.go [hunk 1]
```diff
@@ -50,22 +49,23 @@ func newHTTPTransport(requestTimeout time.Duration) HTTPTransport {
 	}
 }
 
-type mysteriumAPI struct {
+// MysteriumApi provides access to mysterium owned central discovery service
+type MysteriumApi struct {
```
Comment (soffokl) [inline market/mysterium/mysterium_api.go]: > type MysteriumApi should be MysteriumAPI (golint)

File: market/proposals/contact.go [hunk 2]
```diff
@@ -15,7 +15,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package dto
+package proposals
```
Comment (soffokl) [inline market/proposals/contact.go]: Golang code style recommends not use plural for package names. 
Let's call it just `proposal`.

File: cmd/di.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ type Dependencies struct {
 	Node *node.Node
 
 	NetworkDefinition    metadata.NetworkDefinition
-	MysteriumClient      server.Client
+	MysteriumApi         *mysterium.MysteriumApi
```
[the remark this answers is not in the document]
Comment (tadovas) [reply cmd/di.go]: 👍

File: market/proposals/contact.go (hunk 2 above)
Comment (tadovas) [reply market/proposals/contact.go]: Good suggestion. But then here comes my question in PR description: which is better proposal.ServiceProposal or market.ServiceProposal ? I would go for the second

Comment (soffokl) [reply market/proposals/contact.go]: Sorry, forget to answer the question. For me, `proposal.ServiceProposal` sounds better.

File: market/mysterium/mysterium_api.go (hunk 1 above)
Comment (tadovas) [reply market/mysterium/mysterium_api.go]: 👍 fixed

File: market/proposals/contact.go (hunk 2 above)
Comment (tadovas) [reply market/proposals/contact.go]: I guess we will vote during standup :)

File: identity/manager_fake.go [hunk 4]
```diff
@@ -27,6 +27,8 @@ type idmFake struct {
 	unlockFails          bool
```
[the remark this answers is not in the document]
Comment (tadovas) [reply identity/manager_fake.go]: Well this is a bit complicated as we need to split dirs - into read only ones (i.e. keystore) and runtime ones which can be ignored. Maybe in different PR. Does it cause any problems now?

[the remark this answers is not in the document]
Comment (soffokl) [reply identity/manager_fake.go]: oh, I see, it was already committed by me. :)
ok, I'll remove it later with another PR.

File: consumer/statistics/reporter_test.go [hunk 5]
```diff
@@ -56,70 +53,67 @@ func mockLocationDetector() location.Location {
...
 	assert.False(t, reporter.started)
 }
 
-func waitFor(f func() bool) error {
-	timeout := time.Now().Add(time.Second)
-	for time.Now().Before(timeout) {
-		if f() {
-			return nil
-		}
+func waitForChannel(ch chan bool, duration time.Duration) error {
```
Comment (zolia) [inline consumer/statistics/reporter_test.go]: We could have this as a general function somewhere in utils.

File: tequilapi/doc.go [hunk 6]
```diff
@@ -15,7 +15,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-// Tequila API
+// Package tequilapi Tequila API
```
Comment (vkuznecovas) [inline tequilapi/doc.go]: can we add a proper description here since we're already changing it?

Comment (tadovas) [reply tequilapi/doc.go]: What do you call proper description? This change was made to make golint happy.

File: consumer/statistics/reporter_test.go (hunk 5 above)
Comment (tadovas) [reply consumer/statistics/reporter_test.go]: Probably. But not worth it at the moment.

File: consumer/statistics/reporter.go [hunk 7]
```diff
@@ -45,14 +44,20 @@ type Retriever interface {
 	Retrieve() consumer.SessionStatistics
 }
 
+// RemoteReporter defines method for sending stats outside
+// TODO probably bad naming needs improvement or better definition of our statistics server
+type RemoteReporter interface {
+	SendSessionStats(session.ID, mysterium.SessionStats, identity.Signer) error
```
[the remark this answers is not in the document]
Comment (tadovas) [reply consumer/statistics/reporter.go]: Done

File: tequilapi/doc.go (hunk 6 above)
Comment (vkuznecovas) [reply tequilapi/doc.go]: Something along the lines of `Package tequilapi contains the Tequila API http server as well as a client for interacting with it`

Comment (tadovas) [reply tequilapi/doc.go]: I wonder how it look in swagger docs. As this first line is important for godoc and for swagger too.

**Pull request merged**